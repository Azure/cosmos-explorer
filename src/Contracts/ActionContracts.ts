/**
 * Data Explorer tab kinds
 */
export enum TabKind {
  SQLDocuments,
  MongoDocuments,
  MongoSchema,
  TableEntities,
  Graph,
  SQLQuery,
  ScaleSettings,
}

/**
 * Data Explorer pane kinds
 */
export enum PaneKind {
  AddCollection,
  CassandraAddCollection,
  DeleteCollection,
  DeleteDatabase,
  GlobalSettings,
  AdHocAccess,
  SwitchDirectory,
}

/**
 * Parameters to pass to DataExplorer in order to have it perform a given action
 */
export interface DataExplorerAction {
  actionType: ActionType | string;
}

/**
 * Open tab action
 */
export interface OpenTab extends DataExplorerAction {
  tabKind: TabKind | string;
}

/**
 * Open collection tab action
 */
export interface OpenCollectionTab extends OpenTab {
  databaseResourceId: string;
  collectionResourceId: string;
}

/**
 * Open query tab action
 */
export interface OpenQueryTab extends OpenCollectionTab {
  query: QueryInfo;
}

/**
 * Query info will be looked at in the order given in this interface (i.e. if text is provided, everything else is ignored)
 */
export interface QueryInfo {
  text?: string;
  partitionKeys?: string[];
}

/**
 * Open pane action
 */
export interface OpenPane extends DataExplorerAction {
  paneKind: PaneKind | string;
}

export interface OpenSampleNotebook extends DataExplorerAction {
  path: string;
}

/**
 * The types of actions that the DataExplorer supports performing upon opening.
 */
export enum ActionType {
  OpenTab,
  OpenCollectionTab,
  OpenPane,
  TransmitCachedData,
  OpenSampleNotebook,
}
