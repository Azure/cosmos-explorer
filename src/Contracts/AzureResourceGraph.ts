export interface QueryRequestOptions {
  $skipToken?: string;
  $top?: number;
  allowPartialScopes: boolean;
  subscriptions?: string[];
}

export interface QueryResponse {
  $skipToken: string;
  count: number;
  data: any;
  resultTruncated: boolean;
  totalRecords: number;
}
