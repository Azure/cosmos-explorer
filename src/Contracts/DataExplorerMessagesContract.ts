import { FabricMessageTypes } from "./FabricMessageTypes";

// This is the current version of these messages
export const DATA_EXPLORER_RPC_VERSION = "3";

// Data Explorer to Fabric
export type DataExploreMessageV3 =
  | {
      type: FabricMessageTypes.Ready;
      id: string;
      params: [string]; // version
    }
  | {
      type: FabricMessageTypes.GetAuthorizationToken;
      id: string;
      params: GetCosmosTokenMessageOptions[];
    }
  | {
      type: FabricMessageTypes.GetAllResourceTokens;
      id: string;
    }
  | {
      type: FabricMessageTypes.OpenSettings;
      settingsId: string;
    };
export interface GetCosmosTokenMessageOptions {
  verb: "connect" | "delete" | "get" | "head" | "options" | "patch" | "post" | "put" | "trace";
  resourceType: "" | "dbs" | "colls" | "docs" | "sprocs" | "pkranges";
  resourceId: string;
}
