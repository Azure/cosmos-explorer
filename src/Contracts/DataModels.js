"use strict";
exports.__esModule = true;
exports.SparkClusterEndpointKind = exports.ConflictResolutionMode = exports.ApiKind = void 0;
var ApiKind;
(function (ApiKind) {
    ApiKind[ApiKind["SQL"] = 0] = "SQL";
    ApiKind[ApiKind["MongoDB"] = 1] = "MongoDB";
    ApiKind[ApiKind["Table"] = 2] = "Table";
    ApiKind[ApiKind["Cassandra"] = 3] = "Cassandra";
    ApiKind[ApiKind["Graph"] = 4] = "Graph";
    ApiKind[ApiKind["MongoDBCompute"] = 5] = "MongoDBCompute";
})(ApiKind = exports.ApiKind || (exports.ApiKind = {}));
var ConflictResolutionMode;
(function (ConflictResolutionMode) {
    ConflictResolutionMode["Custom"] = "Custom";
    ConflictResolutionMode["LastWriterWins"] = "LastWriterWins";
})(ConflictResolutionMode = exports.ConflictResolutionMode || (exports.ConflictResolutionMode = {}));
var SparkClusterEndpointKind;
(function (SparkClusterEndpointKind) {
    SparkClusterEndpointKind["SparkUI"] = "SparkUI";
    SparkClusterEndpointKind["HistoryServerUI"] = "HistoryServerUI";
    SparkClusterEndpointKind["Livy"] = "Livy";
})(SparkClusterEndpointKind = exports.SparkClusterEndpointKind || (exports.SparkClusterEndpointKind = {}));
