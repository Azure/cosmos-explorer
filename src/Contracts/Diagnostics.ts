/**
 * Trace level.
 */
export enum LogEntryLevel {
  /**
   * Custom events.
   */
  Custom = -2,
  /**
   * Debug level.
   */
  Debug = -1,
  /**
   * Verbose level.
   */
  Verbose = 0,
  /**
   * Warning level.
   */
  Warning = 1,
  /**
   * Error level.
   */
  Error = 2,
}
/**
 * Schema of a log entry.
 */
export interface LogEntry {
  /**
   * Timestamp
   */
  timestamp: number;
  /**
   * Level
   */
  level: LogEntryLevel;
  /**
   * Portal, etc
   */
  area: string;
  /**
   * The message to be logged.
   */
  message: string;
  /**
   * The message code.
   */
  code: number;
  /**
   * Any additional data to be logged.
   */
  args?: any[];
}
export type LogMessage = string | Error;
