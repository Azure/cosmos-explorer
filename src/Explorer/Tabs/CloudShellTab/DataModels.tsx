/**
 * Copyright (c) Microsoft Corporation.  All rights reserved.
 */

export const enum OsType {
    Linux = "linux",
    Windows = "windows"
}

export const enum ShellType {
    Bash = "bash",
    PowerShellCore = "pwsh"
}
  
export const enum NetworkType {
    Default = "Default",
    Isolated = "Isolated"
}

export const enum SessionType {
    Mounted = "Mounted",
    Ephemeral = "Ephemeral"
}

export type Settings = {
    location: string;
    sessionType: SessionType;
    osType: OsType;
};

export type ProvisionConsoleResponse = {
    properties: {
        osType: OsType;
        provisioningState: string;
        uri: string;
    };
};

export type Authorization = {
    token: string;
};

export type ConnectTerminalResponse = {
    id: string;
    idleTimeout: string;
    rootDirectory: string;
    socketUri: string;
    tokenUpdated: boolean;
};


