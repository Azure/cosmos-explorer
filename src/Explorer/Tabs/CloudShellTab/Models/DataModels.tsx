export const enum OsType {
  Linux = "linux",
  Windows = "windows",
}

export const enum ShellType {
  Bash = "bash",
  PowerShellCore = "pwsh",
}

export const enum NetworkType {
  Default = "Default",
  Isolated = "Isolated",
}

/**
 * Azure CloudShell session types:
 * - Mounted: Sessions with persistent storage via an Azure File Share mount.
 *   Files and configurations are preserved between sessions, allowing for
 *   continuity of work across multiple CloudShell sessions.
 *
 * - Ephemeral: Temporary sessions without persistent storage.
 *   All files and changes are discarded when the session ends.
 *   These sessions start faster but don't retain user data.
 *
 * The session type affects resource allocation, startup time,
 * and whether user files/configurations persist between sessions.
 */
export const enum SessionType {
  Mounted = "Mounted",
  Ephemeral = "Ephemeral",
}

export type CloudShellSettings = {
  properties: UserSettingProperties;
};

export type UserSettingProperties = {
  networkType: string;
  preferredLocation: string;
  preferredOsType: OsType;
  preferredShellType: ShellType;
  userSubscription: string;
  sessionType: SessionType;
  vnetSettings: object;
};

export type ProvisionConsoleResponse = {
  properties: {
    osType: OsType;
    provisioningState: string;
    uri: string;
  };
};

export type Authorization = {
  token: string;
};

export type ConnectTerminalResponse = {
  id: string;
  idleTimeout: string;
  rootDirectory: string;
  socketUri: string;
  tokenUpdated: boolean;
};

export type ProviderAuthorization = {
  applicationId: string;
  roleDefinitionId: string;
};

export type ProviderResourceType = {
  resourceType: string;
  locations: string[];
  apiVersions: string[];
  defaultApiVersion?: string;
  capabilities?: string;
};

export type RegistrationState = "Registered" | "NotRegistered" | "Registering" | "Unregistering";
export type RegistrationPolicy = "RegistrationRequired" | "RegistrationOptional";

export type CloudShellProviderInfo = {
  id: string;
  namespace: string;
  authorizations?: ProviderAuthorization[];
  resourceTypes: ProviderResourceType[];
  registrationState: RegistrationState;
  registrationPolicy: RegistrationPolicy;
};
