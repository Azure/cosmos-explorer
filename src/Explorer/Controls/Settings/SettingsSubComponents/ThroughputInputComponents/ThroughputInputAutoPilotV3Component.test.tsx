import { shallow } from "enzyme";
import React from "react";
import * as DataModels from "../../../../../Contracts/DataModels";
import {
  ThroughputInputAutoPilotV3Component,
  ThroughputInputAutoPilotV3Props,
} from "./ThroughputInputAutoPilotV3Component";

describe("ThroughputInputAutoPilotV3Component", () => {
  const baseProps: ThroughputInputAutoPilotV3Props = {
    databaseAccount: {} as DataModels.DatabaseAccount,
    databaseName: "test",
    collectionName: "test",
    wasAutopilotOriginallySet: false,
    throughput: 100,
    throughputBaseline: 100,
    onThroughputChange: undefined,
    minimum: 10000,
    maximum: 400,
    step: 100,
    usageSizeInKB: 10000,
    isEnabled: true,
    isEmulator: false,
    spendAckChecked: false,
    spendAckId: "spendAckId",
    spendAckText: "spendAckText",
    spendAckVisible: false,
    showAsMandatory: true,
    isFixed: false,
    isFreeTierAccount: false,
    label: "label",
    infoBubbleText: "infoBubbleText",
    canExceedMaximumValue: true,
    onAutoPilotSelected: undefined,
    isAutoPilotSelected: false,
    maxAutoPilotThroughput: 4000,
    maxAutoPilotThroughputBaseline: 4000,
    onMaxAutoPilotThroughputChange: undefined,
    onScaleSaveableChange: () => {
      return;
    },
    onScaleDiscardableChange: () => {
      return;
    },
    instantMaximumThroughput: 5000,
    softAllowedMaximumThroughput: 1000000,
    isGlobalSecondaryIndex: false,
  };

  it("throughput input visible", () => {
    const wrapper = shallow(<ThroughputInputAutoPilotV3Component {...baseProps} />);
    const throughputComponent = wrapper.instance() as ThroughputInputAutoPilotV3Component;
    expect(throughputComponent.hasProvisioningTypeChanged()).toEqual(false);
    expect(throughputComponent.overrideWithProvisionedThroughputSettings()).toEqual(false);
    expect(throughputComponent.overrideWithAutoPilotSettings()).toEqual(false);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists("#throughputInput")).toEqual(true);
    expect(wrapper.exists("#autopilotInput")).toEqual(false);
    expect(wrapper.exists("#throughputSpendElement")).toEqual(true);
  });

  it("autopilot input visible", () => {
    const newProps = { ...baseProps, isAutoPilotSelected: true };
    const wrapper = shallow(<ThroughputInputAutoPilotV3Component {...newProps} />);
    const throughputComponent = wrapper.instance() as ThroughputInputAutoPilotV3Component;
    expect(throughputComponent.hasProvisioningTypeChanged()).toEqual(true);
    expect(throughputComponent.overrideWithProvisionedThroughputSettings()).toEqual(true);
    expect(throughputComponent.overrideWithAutoPilotSettings()).toEqual(false);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists("#autopilotInput")).toEqual(true);
    expect(wrapper.exists("#throughputInput")).toEqual(false);
    expect(wrapper.exists("#manualToAutoscaleDisclaimerElement")).toEqual(true);

    wrapper.setProps({ wasAutopilotOriginallySet: true });
    wrapper.update();
    expect(wrapper.exists("#throughputSpendElement")).toEqual(true);
  });

  it("spendAck checkbox visible", () => {
    const newProps = { ...baseProps, spendAckVisible: true };
    const wrapper = shallow(<ThroughputInputAutoPilotV3Component {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists("#spendAckCheckBox")).toEqual(true);
  });

  it("scale saveable and discardable are set", () => {
    let throughputComponent = new ThroughputInputAutoPilotV3Component(baseProps);
    let isComponentDirtyResult = throughputComponent.IsComponentDirty();
    expect(isComponentDirtyResult.isSaveable).toEqual(false);
    expect(isComponentDirtyResult.isDiscardable).toEqual(false);

    const newProps = { ...baseProps, throughput: 1000000 };
    throughputComponent = new ThroughputInputAutoPilotV3Component(newProps);
    isComponentDirtyResult = throughputComponent.IsComponentDirty();
    expect(isComponentDirtyResult.isSaveable).toEqual(true);
    expect(isComponentDirtyResult.isDiscardable).toEqual(true);
  });
});
