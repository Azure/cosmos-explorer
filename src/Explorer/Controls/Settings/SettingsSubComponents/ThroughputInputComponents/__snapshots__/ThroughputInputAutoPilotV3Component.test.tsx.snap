// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThroughputInputAutoPilotV3Component autopilot input visible 1`] = `
<Stack
  tokens={
    Object {
      "childrenGap": 10,
    }
  }
>
  <StyledMessageBar
    messageBarIconProps={
      Object {
        "className": "messageBarWarningIcon",
        "iconName": "WarningSolid",
      }
    }
  >
    <Text
      styles={
        Object {
          "root": Object {
            "fontSize": 14,
          },
        }
      }
    >
      Your bill will be affected as you update your throughput settings. Please review the updated cost estimate below before saving your changes
    </Text>
  </StyledMessageBar>
  <Stack>
    <StyledLabelBase
      id="settingsV2RadioButtonLabelId"
    >
      <ToolTipLabelComponent
        label="label"
        toolTipElement={
          <Text
            styles={
              Object {
                "root": Object {
                  "fontSize": 14,
                },
              }
            }
          >
            infoBubbleText
          </Text>
        }
      />
    </StyledLabelBase>
    <StyledMessageBar
      messageBarIconProps={
        Object {
          "className": "messageBarInfoIcon",
          "iconName": "InfoSolid",
        }
      }
      styles={
        Object {
          "root": Object {
            "backgroundColor": "white",
            "marginTop": "5px",
          },
          "text": Object {
            "fontSize": 14,
          },
        }
      }
    >
      <Text
        id="manualToAutoscaleDisclaimerElement"
        styles={
          Object {
            "root": Object {
              "fontSize": 14,
            },
          }
        }
      >
        The starting autoscale max RU/s will be determined by the system, based on the current manual throughput settings and storage of your resource. After autoscale has been enabled, you can change the max RU/s.
         
        <a
          href="https://aka.ms/cosmos-autoscale-migration"
        >
          Learn more
        </a>
      </Text>
    </StyledMessageBar>
    <StyledChoiceGroup
      ariaLabelledBy="settingsV2RadioButtonLabelId"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "true",
            "text": "Autoscale",
          },
          Object {
            "key": "false",
            "text": "Manual",
          },
        ]
      }
      required={true}
      selectedKey="true"
      styles={
        Object {
          "flexContainer": Array [
            Object {
              "selectors": Object {
                ".ms-ChoiceField-field.is-checked::after": Object {
                  "borderColor": undefined,
                },
                ".ms-ChoiceField-field.is-checked::before": Object {
                  "borderColor": undefined,
                },
                ".ms-ChoiceField-wrapper label": Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "padding": "2px 5px",
                  "whiteSpace": "nowrap",
                },
              },
            },
          ],
        }
      }
    />
  </Stack>
  <Text>
    Provision maximum RU/s required by this resource. Estimate your required RU/s with
    <StyledLinkBase
      href="https://cosmos.azure.com/capacitycalculator/"
      target="_blank"
    >
       capacity calculator
    </StyledLinkBase>
  </Text>
  <StyledTextFieldBase
    disabled={true}
    id="autopilotInput"
    key="auto pilot throughput input"
    label="Max RU/s"
    min={4000}
    onChange={[Function]}
    required={true}
    step={1000}
    styles={
      Object {
        "fieldGroup": Object {
          "borderColor": "",
          "height": 25,
          "selectors": Object {
            ":disabled": Object {
              "backgroundColor": undefined,
              "borderColor": undefined,
            },
          },
          "width": 300,
        },
      }
    }
    type="number"
    value=""
  />
</Stack>
`;

exports[`ThroughputInputAutoPilotV3Component spendAck checkbox visible 1`] = `
<Stack
  tokens={
    Object {
      "childrenGap": 10,
    }
  }
>
  <Stack>
    <StyledLabelBase
      id="settingsV2RadioButtonLabelId"
    >
      <ToolTipLabelComponent
        label="label"
        toolTipElement={
          <Text
            styles={
              Object {
                "root": Object {
                  "fontSize": 14,
                },
              }
            }
          >
            infoBubbleText
          </Text>
        }
      />
    </StyledLabelBase>
    <StyledChoiceGroup
      ariaLabelledBy="settingsV2RadioButtonLabelId"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "true",
            "text": "Autoscale",
          },
          Object {
            "key": "false",
            "text": "Manual",
          },
        ]
      }
      required={true}
      selectedKey="false"
      styles={
        Object {
          "flexContainer": Array [
            Object {
              "selectors": Object {
                ".ms-ChoiceField-field.is-checked::after": Object {
                  "borderColor": "",
                },
                ".ms-ChoiceField-field.is-checked::before": Object {
                  "borderColor": "",
                },
                ".ms-ChoiceField-wrapper label": Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "padding": "2px 5px",
                  "whiteSpace": "nowrap",
                },
              },
            },
          ],
        }
      }
    />
  </Stack>
  <Stack
    tokens={
      Object {
        "childrenGap": 5,
      }
    }
  >
    <Text>
      Estimate your required throughput with
      <StyledLinkBase
        href="https://cosmos.azure.com/capacitycalculator/"
        target="_blank"
      >
         capacity calculator
         
        <FontIcon
          iconName="NavigateExternalInline"
        />
      </StyledLinkBase>
    </Text>
    <StyledTextFieldBase
      disabled={false}
      id="throughputInput"
      key="provisioned throughput input"
      min={10000}
      onChange={[Function]}
      required={true}
      step={100}
      styles={
        Object {
          "fieldGroup": Object {
            "borderColor": "",
            "height": 25,
            "selectors": Object {
              ":disabled": Object {
                "backgroundColor": undefined,
                "borderColor": undefined,
              },
            },
            "width": 300,
          },
        }
      }
      type="number"
      value="100"
    />
    <Stack
      styles={
        Object {
          "root": Object {
            "width": 600,
          },
        }
      }
      tokens={
        Object {
          "childrenGap": 10,
        }
      }
    >
      <StyledWithViewportComponent
        columns={
          Array [
            Object {
              "fieldName": "costType",
              "isResizable": true,
              "key": "costType",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
            Object {
              "fieldName": "hourly",
              "isResizable": true,
              "key": "hourly",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "Hourly",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
            Object {
              "fieldName": "daily",
              "isResizable": true,
              "key": "daily",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "Daily",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
            Object {
              "fieldName": "monthly",
              "isResizable": true,
              "key": "monthly",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "Monthly",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
          ]
        }
        disableSelectionZone={true}
        items={
          Array [
            Object {
              "costType": <Text>
                Current Cost
              </Text>,
              "daily": <Text>
                $
                 
                0.19
              </Text>,
              "hourly": <Text>
                $
                 
                0.0080
              </Text>,
              "monthly": <Text>
                $
                 
                5.84
              </Text>,
            },
          ]
        }
        layoutMode={1}
        onRenderRow={[Function]}
        selectionMode={0}
      />
      <Text
        id="throughputSpendElement"
      >
        (
        regions: 
         
        1
        , 
        100
         RU/s, 
        $
        0.00008
        /RU)
      </Text>
      <Text>
        <em>
          This cost is an estimate and may vary based on the regions where your account is deployed and potential discounts applied to your account
        </em>
      </Text>
    </Stack>
    <StyledCheckboxBase
      checked={false}
      id="spendAckCheckBox"
      label="spendAckText"
      onChange={[Function]}
      styles={
        Object {
          "label": Object {
            "margin": 0,
            "padding": "2 0 2 0",
          },
          "text": Object {
            "fontSize": 12,
          },
        }
      }
    />
    <br />
  </Stack>
</Stack>
`;

exports[`ThroughputInputAutoPilotV3Component throughput input visible 1`] = `
<Stack
  tokens={
    Object {
      "childrenGap": 10,
    }
  }
>
  <Stack>
    <StyledLabelBase
      id="settingsV2RadioButtonLabelId"
    >
      <ToolTipLabelComponent
        label="label"
        toolTipElement={
          <Text
            styles={
              Object {
                "root": Object {
                  "fontSize": 14,
                },
              }
            }
          >
            infoBubbleText
          </Text>
        }
      />
    </StyledLabelBase>
    <StyledChoiceGroup
      ariaLabelledBy="settingsV2RadioButtonLabelId"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "true",
            "text": "Autoscale",
          },
          Object {
            "key": "false",
            "text": "Manual",
          },
        ]
      }
      required={true}
      selectedKey="false"
      styles={
        Object {
          "flexContainer": Array [
            Object {
              "selectors": Object {
                ".ms-ChoiceField-field.is-checked::after": Object {
                  "borderColor": "",
                },
                ".ms-ChoiceField-field.is-checked::before": Object {
                  "borderColor": "",
                },
                ".ms-ChoiceField-wrapper label": Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "padding": "2px 5px",
                  "whiteSpace": "nowrap",
                },
              },
            },
          ],
        }
      }
    />
  </Stack>
  <Stack
    tokens={
      Object {
        "childrenGap": 5,
      }
    }
  >
    <Text>
      Estimate your required throughput with
      <StyledLinkBase
        href="https://cosmos.azure.com/capacitycalculator/"
        target="_blank"
      >
         capacity calculator
         
        <FontIcon
          iconName="NavigateExternalInline"
        />
      </StyledLinkBase>
    </Text>
    <StyledTextFieldBase
      disabled={false}
      id="throughputInput"
      key="provisioned throughput input"
      min={10000}
      onChange={[Function]}
      required={true}
      step={100}
      styles={
        Object {
          "fieldGroup": Object {
            "borderColor": "",
            "height": 25,
            "selectors": Object {
              ":disabled": Object {
                "backgroundColor": undefined,
                "borderColor": undefined,
              },
            },
            "width": 300,
          },
        }
      }
      type="number"
      value="100"
    />
    <Stack
      styles={
        Object {
          "root": Object {
            "width": 600,
          },
        }
      }
      tokens={
        Object {
          "childrenGap": 10,
        }
      }
    >
      <StyledWithViewportComponent
        columns={
          Array [
            Object {
              "fieldName": "costType",
              "isResizable": true,
              "key": "costType",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
            Object {
              "fieldName": "hourly",
              "isResizable": true,
              "key": "hourly",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "Hourly",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
            Object {
              "fieldName": "daily",
              "isResizable": true,
              "key": "daily",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "Daily",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
            Object {
              "fieldName": "monthly",
              "isResizable": true,
              "key": "monthly",
              "maxWidth": 200,
              "minWidth": 100,
              "name": "Monthly",
              "styles": Object {
                "root": Object {
                  "selectors": Object {
                    ":hover": Object {
                      "background": "transparent",
                    },
                  },
                },
              },
            },
          ]
        }
        disableSelectionZone={true}
        items={
          Array [
            Object {
              "costType": <Text>
                Current Cost
              </Text>,
              "daily": <Text>
                $
                 
                0.19
              </Text>,
              "hourly": <Text>
                $
                 
                0.0080
              </Text>,
              "monthly": <Text>
                $
                 
                5.84
              </Text>,
            },
          ]
        }
        layoutMode={1}
        onRenderRow={[Function]}
        selectionMode={0}
      />
      <Text
        id="throughputSpendElement"
      >
        (
        regions: 
         
        1
        , 
        100
         RU/s, 
        $
        0.00008
        /RU)
      </Text>
      <Text>
        <em>
          This cost is an estimate and may vary based on the regions where your account is deployed and potential discounts applied to your account
        </em>
      </Text>
    </Stack>
    <br />
  </Stack>
</Stack>
`;
